Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -433,17 +433,19 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
                     });
                 }
             });
-            let result: number = this.printBuildSummary(error, stdout, stderr);
-            if (spawnResult === -1) {
-                result = -1;
-            }
+            const result: number = this.printBuildSummary(error, stdout, stderr, spawnResult);
             this.closeEmitter.fire(result);
         } catch {
             this.closeEmitter.fire(-1);
         }
     }
 
-    private printBuildSummary(error: string, stdout: string, stderr: string): number {
+    private printBuildSummary(error: string, stdout: string, stderr: string, spawnResult: number): number {
+        if (spawnResult === -1) {
```

Comment:
Failure is defined as any non-zero result.  line 431 could have returned `1` or `100` or any other number.  The only time it indicates success is when it is `0`.